src/main/java/ch/uzh/ifi/hase/soprafs21/controller/MessageApiController.java

```diff
@@ -12,13 +29,35 @@
     private final NativeWebRequest request;
 
     @org.springframework.beans.factory.annotation.Autowired
-    public MessageApiController(NativeWebRequest request) {
+    public MessageApiController(NativeWebRequest request, UserService userService, ChatService chatService) {
         this.request = request;
+        this.userService = userService;
+        this.chatService = chatService;
     }
 
+    private final UserService userService;
+    private final ChatService chatService;
+
     @Override
     public Optional<NativeWebRequest> getRequest() {
         return Optional.ofNullable(request);
     }
 
+    @Override
+    public ResponseEntity<Void> messagePost(@ApiParam(value = "A message that needs to be created" ,required=true )  @Valid @RequestBody ChatMessageDto chatMessageDto) throws Exception {
+        Long senderId = chatMessageDto.getSender().getId();
+        Long receiverId = chatMessageDto.getReceiver().getId();
```

Comment:
They are not full User objects but UserOverviewDto (which is already simplified ). We could create a separate obj for a messege POST as well.

Reply:
We will not need all the info in UserOverviewDto with each message when we get the full conversation, and even less so when we post a new message.

I think this should only contain the message and sender/recipient IDs. But we can also adjust it as needed when we implement the corresponding frontend parts and leave it like this for now.

Reply:
I see your point that we do not need to have some of the fields of receiver/sender  when we get a message in our current implementation. But I did it this way so that the code is more reusable. In case we decide that we need more info of the message sender, for example. 

For the POST I will create the DTO with receiver and sender IDs only.